src/java.xml.crypto/share/classes/javax/xml/crypto/dsig/spec/RSAPSSParameterSpec.java

```diff
@@ -0,0 +1,145 @@
+import java.security.spec.MGF1ParameterSpec;
+import java.security.spec.PSSParameterSpec;
+import java.util.Objects;
+
+/**
+ * Parameters for the <a href="http://www.w3.org/2007/05/xmldsig-more#rsa-pss">
+ * XML Signature RSASSA-PSS Algorithm</a>. The parameters are expressed as a
+ * {@link PSSParameterSpec} object encapsulated.
+ * <p>
+ * The XML Schema Definition is defined as:
+ * <pre><code>
+ *    &lt;xs:element name="RSAPSSParams" type="pss:RSAPSSParamsType"&gt;
+ *        &lt;xs:annotation&gt;
+ *            &lt;xs:documentation&gt;
+ *    Top level element that can be used in xs:any namespace="#other"
+ *    wildcard of ds:SignatureMethod content.
+ *            &lt;/xs:documentation&gt;
+ *        &lt;/xs:annotation&gt;
+ *    &lt;/xs:element&gt;
+ *    &lt;xs:complexType name="RSAPSSParamsType"&gt;
+ *        &lt;xs:sequence&gt;
+ *            &lt;xs:element ref="ds:DigestMethod" minOccurs="0"/&gt;
+ *            &lt;xs:element name="MaskGenerationFunction"
+ *               type="pss:MaskGenerationFunctionType" minOccurs="0"/&gt;
+ *            &lt;xs:element name="SaltLength" type="xs:int"
+ *               minOccurs="0"/&gt;
+ *            &lt;xs:element name="TrailerField" type="xs:int"
+ *               minOccurs="0"/&gt;
+ *        &lt;/xs:sequence&gt;
+ *    &lt;/xs:complexType&gt;
+ *    &lt;xs:complexType name="MaskGenerationFunctionType"&gt;
+ *        &lt;xs:sequence&gt;
+ *            &lt;xs:element ref="ds:DigestMethod" minOccurs="0"/&gt;
+ *        &lt;/xs:sequence&gt;
+ *        &lt;xs:attribute name="Algorithm" type="xs:anyURI"
+ *           default="http://www.w3.org/2007/05/xmldsig-more#MGF1"/&gt;
+ *    &lt;/xs:complexType&gt;
+ * </code></pre>
+ *
+ * @implSpec
+ * If the parameter is not specified when using the RSASSA-PSS signature
+ * algorithm, the default parameter is used, which uses SHA-256 as the
+ * {@code DigestMethod}, MGF1 with SHA-256 as the
+ * {@code MaskGenerationFunction}, 32 as {@code SaltLength}, and 1 as
+ * {@code TrailerField}. This is equivalent to the parameter-less signature
+ * method as defined by http://www.w3.org/2007/05/xmldsig-more#sha256-rsa-MGF1.
+ *
+ * @since 17
+ * @see SignatureMethod
+ * @see <a href="https://www.ietf.org/rfc/rfc6931.txt">RFC 6931</a>
+ */
+public final class RSAPSSParameterSpec implements SignatureMethodParameterSpec {
+
+    private final PSSParameterSpec spec;
+
+    /**
+     * Creates a new {@code RSAPSSParameterSpec} object with the specified
+     * {@link PSSParameterSpec} object encapsulated.
+     *
+     * @param spec the input {@code PSSParameterSpec} to be encapsulated
+     */
+    public RSAPSSParameterSpec(PSSParameterSpec spec) {
+        this.spec = Objects.requireNonNull(spec);
+    }
+
+    /**
+     * Returns the encapsulated {@code PSSParameterSpec} object.
+     *
+     * @return the encapsulated {@code PSSParameterSpec} object
+     */
+    public PSSParameterSpec getPSSParameterSpec() {
+        return spec;
+    }
+
+    @Override
```

Comment:
Since you are overriding `Object.hashCode` and `equals`, I think you should document the specification for that.